go/vrf/ecvrf.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vrf
+
+import (
+	"crypto/elliptic"
+	"math/big"
+)
+
+// PublicKey holds a public VRF key.
+type PublicKey struct {
+	elliptic.Curve
+	X, Y *big.Int
+}
+
+// PrivateKey holds a private VRF key.
+type PrivateKey struct {
+	PublicKey
+	d *big.Int
+}
+
+// Public returns the public key corresponding to priv.
+func (priv *PrivateKey) Public() *PublicKey {
+	return &priv.PublicKey
+}
+
+func NewKey(curve elliptic.Curve, sk []byte) *PrivateKey {
+	Yx, Yy := curve.ScalarBaseMult(sk)
```

Comment:
Why Yx and Yy and not just X, Y? It is not clear from https://golang.org/src/crypto/elliptic/elliptic.go?s=7072:7143#L269.

Reply:
done